es6-wrapped-promise.js

```diff
@@ -0,0 +1,35 @@
+'use strict';
+
+module.exports = (Promise, ensureAslWrapper) => {
+  return class WrappedPromise extends Promise {
+    constructor(executor) {
+      var context, args;
+      super(wrappedExecutor);
+      var promise = this;
+
+      try {
+        executor.apply(context, args);
+      } catch (err) {
+        args[1](err)
+      }
+
+      return promise;
+      function wrappedExecutor(resolve, reject) {
```

Comment:
In case we ever need to make a change to this function, I'd probably add a comment noting that this is a duplicate of the function in index.js. And I'd also make a similar comment by the function in index.js. Else they might deviate by a mistake at some point